components/script/dom/readablestreambyobrequest.rs

```diff
@@ -3,37 +3,72 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 use dom_struct::dom_struct;
+use js::gc::CustomAutoRooterGuard;
+use js::jsapi::Heap;
+use js::typedarray::{ArrayBufferView, ArrayBufferViewU8};
 
+use super::bindings::buffer_source::{BufferSource, HeapBufferSource};
 use crate::dom::bindings::codegen::Bindings::ReadableStreamBYOBRequestBinding::ReadableStreamBYOBRequestMethods;
 use crate::dom::bindings::import::module::{Error, Fallible};
 use crate::dom::bindings::reflector::Reflector;
+use crate::dom::bindings::root::MutNullableDom;
+use crate::dom::readablebytestreamcontroller::ReadableByteStreamController;
 use crate::script_runtime::JSContext as SafeJSContext;
 
 /// <https://streams.spec.whatwg.org/#readablestreambyobrequest>
 #[dom_struct]
 pub(crate) struct ReadableStreamBYOBRequest {
     reflector_: Reflector,
+    controller: MutNullableDom<ReadableByteStreamController>,
+    #[ignore_malloc_size_of = "mozjs"]
+    view: HeapBufferSource<ArrayBufferViewU8>,
 }
 
 impl ReadableStreamBYOBRequestMethods<crate::DomTypeHolder> for ReadableStreamBYOBRequest {
     /// <https://streams.spec.whatwg.org/#rs-byob-request-view>
     fn GetView(&self, _cx: SafeJSContext) -> Option<js::typedarray::ArrayBufferView> {
-        // TODO
-        None
+        self.view.buffer_to_option()
     }
 
     /// <https://streams.spec.whatwg.org/#rs-byob-request-respond>
-    fn Respond(&self, _bytes_written: u64) -> Fallible<()> {
-        // TODO
-        Err(Error::NotFound)
+    fn Respond(&self, bytes_written: u64) -> Fallible<()> {
+        // If this.[[controller]] is undefined, throw a TypeError exception.
+        if self.controller.get().is_none() {
+            return Err(Error::Type("controller is undefined".to_owned()));
+        }
+        // If ! IsDetachedBuffer(this.[[view]].[[ArrayBuffer]]) is true, throw a TypeError exception.
+        if self.view.is_detached_buffer() {
+            return Err(Error::Type("buffer is detached".to_owned()));
+        }
+
+        // Assert: this.[[view]].[[ByteLength]] > 0.
+        assert!(self.view.byte_length() > 0);
+
+        // Assert: this.[[view]].[[ViewedArrayBuffer]].[[ByteLength]] > 0.
+        assert!(self.view.viewed_buffer_byte_length() > 0);
+
+        // Perform ? ReadableByteStreamControllerRespond(this.[[controller]], bytesWritten).
+        self.controller.get().unwrap().respond(bytes_written)
```

Comment:
We can replace this with an let else with an unreachable at the top of the function.